pkg/txnutil/gc/gc_manager.go

```diff
@@ -109,20 +107,32 @@ func (m *gcManager) CheckStaleCheckpointTs(
 	ctx context.Context, changefeedID model.ChangeFeedID, checkpointTs model.Ts,
 ) error {
 	gcSafepointUpperBound := checkpointTs - 1
-	if m.isTiCDCBlockGC {
-		pdTime, err := m.pdClock.CurrentTime()
-		// TODO: should we return err here, or just log it?
-		if err != nil {
-			return errors.Trace(err)
-		}
-		if pdTime.Sub(oracle.GetTimeFromTS(gcSafepointUpperBound)) > time.Duration(m.gcTTL)*time.Second {
```

Comment:
why do you remove this?

Reply:
After adopting the new algorithm, this function will no longer accept safepoint ts with a lag of more than 24 hours relative to the current pd time, as they will be disregarded by the ignoredFailedChangeFeedWhenGC function.